src/upsell_panda.js

```diff
@@ -48,19 +48,21 @@ function generatePromotionContent(promotion) {
   </div>`;
 }
 
-function renderPromotionInContainer(promotion) {
+function renderPromotionInContainer(promotion, fadeIn) {
   var bannerContainer = document.getElementsByClassName('__promotions_container')[0];
 
   var promotionHolder = document.createElement('span');
   promotionHolder.innerHTML = generatePromotionContent(promotion);
-  promotionHolder.classList = '__shown __promotion';
+  bannerContainer.classList = `__promotions_container ${fadeIn ? '__fade_in' : ''}`;
   bannerContainer.innerHTML = '';
   bannerContainer.appendChild(promotionHolder);
 }
 
 let intervalTracker = 0;
 let intervalID = null;
-function turnOnCountdown(promotion) {
+function insertAndAnimatePromotion(promotion, fadeIn) {
```

Comment:
We will re-render all promotions every second regardless of if they countdown timer or not. This has 2 benefits:
1) On first render we animate fade in by adding `__fade_in` class. On second+ render we remove `__fade_in` class, which will allow us to use `__fade_in` to trigger animation on next promotion render.
2) It's simpler and removes an `if (hasCountdown)` check.